File: April-18/16-04-18/python/01-Edge-Existence.py
n, m = map(int, input().split())
n += 5
m += 5
adj = [[0]*n for i in range(m)]

for i in range(0,m-5):
    a, b = map(int,input().split())
    adj[a][b] = 1

for _ in range(int(input())):
    a, b = map(int,input().split())
    if(adj[a][b] == 1):
        print("YES")
    else:
        print("NO")

File: April-18/16-04-18/java/01-Edge-Existence.java

import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n, m;
        n = input.nextInt();
        m = input.nextInt();
        int adj[][] = new int[n+5][m+5];
        for(int i = 0; i < m; i++)
        {
            int a, b;
            a = input.nextInt();
            b = input.nextInt();
            adj[a][b] = 1;
        }
        int q = input.nextInt();
        while(q-- > 0)
        {
            int a, b;
            a = input.nextInt();
            b = input.nextInt();
            if(adj[a][b] == 1)
                System.out.println("YES");
            else
                System.out.println("NO");
            
        }
    }
}
